### PR TITLE
Refactor useTurns to extract the logic from Recoil code

### PR DESCRIPTION
**Pull request by etrobert**

### CODE REVIEW

File: src/useTurns.ts [hunk 1]
```diff
@@ -30,38 +20,72 @@ const movePosition = (position: Position, direction: Direction) => {
...
 
-  const players = useRecoilValue(playersState);
+        Array.from(newPlayersPositions).map(([playerId, position]) =>
+          set(playerPositionState(playerId), position)
+        );
+
+        players.forEach((playerId) =>
+          set(playerIntentsState(playerId), (intents) => {
+            if (intents.length === 0) return [];
+            const [_, ...rest] = intents;
```
Comment (justinline) [inline src/useTurns.ts]: suggestion: `...queuedIntents`

File: src/useTurns.ts [hunk 2]
```diff
@@ -30,38 +20,72 @@ const movePosition = (position: Position, direction: Direction) => {
...
+          })
+        );
+      }
+  );
 
   useInterval(() => {
-    resetHitPosition();
-    players.forEach(takeTurn);
+    recoilProcessTurn();
   }, 1000);
```
Comment (justinline) [inline src/useTurns.ts]: suggestion: `const turnSpeed = 1000` 😇

**Review by justinline — COMMENTED** [review]
Finished 🏁

File: src/useTurns.ts (hunk 1 above)
Comment (etrobert) [reply src/useTurns.ts]: Used lodash's `drop`

https://docs-lodash.com/v4/drop/

**Pull request merged**